File: c/2894.c

int differenceOfSums(int n, int m)
{
  int divisible_number = 0;
  int not_divisible_number = 0;

  for (int i = 1; i < n + 1; i++)
  {
    if (i % m == 0)
    {
      divisible_number += i;
    }
    else
    {
      not_divisible_number += i;
    }
  }

  return not_divisible_number - divisible_number;
}

File: go/2894.go
package main

func differenceOfSums(n int, m int) int {
	var divisible_by_m int
	var not_divisible_by_m int

	for i := 1; i < n+1; i++ {
		if i%m == 0 {
			divisible_by_m += i
		} else {
			not_divisible_by_m += i
		}
	}

	return not_divisible_by_m - divisible_by_m
}
